Bourreau/spec/boutiques/boutiques_tester_spec.rb

```diff
@@ -0,0 +1,389 @@
+        expect( @task.save_results ).to be true
+        # The file should now exist in the data_provider
+        expect( @task.userfile_exists(@userfileClass, @reqOutfileProps) ).to be true
+        # Create a new task and have it save results as well (with the same name for the output file)
+        task2        = CbrainTask::BoutiquesTest.new
+        task2.params = {:ro => @fname, :interface_userfile_ids => [Userfile.all.first.id]}
+        task2.user_id, task2.group_id = UID, GID
+        # Check that there are no files that match the renamed filename pattern
+        renamedFileRegex = /^#{@fname.split(".")[0]}-.*-\d{9}\.txt$/
+        expect( Userfile.all.any? { |f| (f.name.to_s =~ renamedFileRegex) == 0 } ).to be false
+        # Saving the second output file should be error free
+        expect( task2.save_results ).to be true
+        # Initial file should still be there with the correct, original name
+        expect( @task.userfile_exists(@userfileClass, @reqOutfileProps) ).to be true
+        expect( Userfile.where( :name => @fname ).first.nil? ).not_to be true
+        # New file should exist and have been renamed appropriately
+        expect( Userfile.all.one? { |f| (f.name.to_s =~ renamedFileRegex) == 0 } ).to be true
+      end
+
+    end # End output file handling tests
+
+    # Test that a Tool Config is created iff both the bourreau and the descriptor specify docker
+    context 'Default ToolConfig creation' do
+
+      before(:each) do
+        # Ensure the Bourreau does not have docker installed by default
+        resource = RemoteResource.current_resource
+        resource.docker_present = false
+        resource.save!
+        # Get the schema and json descriptor
+        schema           = SchemaTaskGenerator.default_schema
+        @descriptor      = File.join(__dir__, TestScriptDescriptor)
+        # Generate a new task class via the Boutiques framework, without integrating it
+        @boutiquesTask   = SchemaTaskGenerator.generate(schema, @descriptor)
+        @boutiquesTask.descriptor["docker-image"] = nil # tool does not use docker by default
+        @task_const_name = "CbrainTask::#{SchemaTaskGenerator.classify(@boutiquesTask.name)}"
+        # Destroy any tools/toolconfigs for the tool, if any exist
+        ToolConfig.where(tool_id: CbrainTask::BoutiquesTest.tool.id).destroy_all rescue nil
+        Tool.where(:cbrain_task_class => @task_const_name).destroy_all rescue nil
+      end
+
+      after(:each) do
+        # Destroy any tools/toolconfigs for the tool, if any exist
+        ToolConfig.where(tool_id: @task_const_name.constantize.tool.id).destroy_all
+        Tool.where(:cbrain_task_class => @task_const_name).destroy_all
+        # Ensure the Bourreau does not have docker installed by default
+        resource = RemoteResource.current_resource
+        resource.docker_present = false
+        resource.save!
+      end
+
+      it "is not done with only docker in task" do
```

Comment:
The way rspec structures these is that the 'it' block is appended to the 'contexts' or 'descriptions'. So it is actually:
"Default ToolConfig creation is not done with _only_ docker in task", which means the same thing. But yours might be clearer.